### PR TITLE
ECR support

### PR DESCRIPTION
**Pull request by trombonekenny**
Investigate using the AWS Elastic Container Registry for grading docker containers.

### CODE REVIEW

**Comment by trombonekenny** [comment]
Using `docker-credential-ecr-login` with a docker config.json with `registry-mirrors: ["our.ecr.link"]` still requires the changing the docker image name from `cs125/quiz` to `our.ecr.link/cs125/quiz` to work. I don't think we can implement ECR without code changing PG, at least some how.

One strategy might be to force the graders to use ECR and only pull. That assumes some other system is keeping containers in the ECR in sync. A bigger strategy has the graders pulling and pushing without something else tracking.

**Comment by mwest1066** [comment]
I like the idea of forcing the graders to use ECR and only pull, and leave it up to PL to keep ECR images in sync. The less code we have in PG the better -- it's really designed to be a small and dumb service.

I don't mind whether we use https://github.com/awslabs/amazon-ecr-credential-helper or `ecr.getAuthorizationToken()` to manage auth. What's your preference? I guess I have a slight preference for calling `ecr.getAuthorizationToken()` ourselves.

Can you make a version of this PR that strips out all the push/mirroring code and just does the dumb thing of forcing it to pull from ECR? I can fairly easily test the performance of that with a large CloudFormation stack of graders on pl-dev.

File: lib/pullImages.js [hunk 1]
```diff
@@ -17,9 +20,21 @@ module.exports = function(callback) {
                 callback(null);
             });
         },
+        (callback) => {
+            if (config.forcedRegistry) {
+                logger.info('Authenticating to docker');
+                dockerUtil.setupDockerAuth((err, auth) => {
```
Comment (mwest1066) [inline lib/pullImages.js:26]: Do the tokens we get from this auth only last 12 hours? If so, we'll need to remember to refresh them.

**Comment by mwest1066** [comment]
If we go with the "dumb grader just pulls from ECR" strategy, then I'm thinking the pull/mirror code in this PR would get merged into PL, probably as an explicit "mirror images" button on the Sync page. Does that sound reasonable to you?

File: lib/pullImages.js (hunk 1 above)
Comment (trombonekenny) [reply lib/pullImages.js:26]: We could be smarter about this, but for now I'm refreshing the auth before each call.

**Pull request merged**